packages/@aws-cdk/aws-kinesisfirehose-destinations/lib/s3-bucket.ts

```diff
@@ -93,19 +35,29 @@ export class S3Bucket implements firehose.IDestination {
       streamId: 'S3Destination',
     }) ?? {};
 
+    const { backupConfig, dependables: backupDependables } = createBackupConfig(scope, role, this.props.s3Backup) ?? {};
     return {
       extendedS3DestinationConfiguration: {
         cloudWatchLoggingOptions: loggingOptions,
         processingConfiguration: createProcessingConfig(scope, role, this.props.processor),
         roleArn: role.roleArn,
+        s3BackupConfiguration: backupConfig,
+        s3BackupMode: this.getS3BackupMode(),
         bufferingHints: createBufferingHints(this.props.bufferingInterval, this.props.bufferingSize),
         bucketArn: this.bucket.bucketArn,
         compressionFormat: this.props.compression?.value,
         encryptionConfiguration: createEncryptionConfig(role, this.props.encryptionKey),
         errorOutputPrefix: this.props.errorOutputPrefix,
         prefix: this.props.dataOutputPrefix,
       },
-      dependables: [bucketGrant, ...(loggingDependables ?? [])],
+      dependables: [bucketGrant, ...(loggingDependables ?? []), ...(backupDependables ?? [])],
     };
   }
+
+  private getS3BackupMode(): string {
+    if (this.props.s3Backup?.bucket || this.props.s3Backup?.mode === BackupMode.ALL) {
+      return 'Enabled';
+    }
+    return 'Disabled';
```

Comment:
I would write this as:

```suggestion
    return this.props.s3Backup?.bucket || this.props.s3Backup?.mode === BackupMode.ALL
      ? 'Enabled'
      : 'Disabled';
```

Reply:
Deja vu! 😜